pkg/sdk/sdk.go

```diff
@@ -113,3 +113,23 @@ func NewPOKTRollSDK(ctx context.Context, config *POKTRollSDKConfig) (POKTRollSDK
 
 	return sdk, nil
 }
+
+// getPubKeyFromAddress returns the public key of the given address.
+// It uses the accountQuerier to get the account if it is not already in the cache.
+// If the account does not have a public key, it returns an error.
+func (sdk *poktrollSDK) getPubKeyFromAddress(
+	ctx context.Context,
+	address string,
+) (cryptotypes.PubKey, error) {
+	if pubKey, ok := sdk.supplierPubKeyCache[address]; ok {
+		return pubKey, nil
+	}
+
+	pubKey, err := sdk.pubKeyClient.GetPubKeyFromAddress(ctx, address)
+	if err != nil {
+		return nil, err
+	}
+
+	sdk.supplierPubKeyCache[address] = pubKey
```

Comment:
The function name and comment is pretty general purpose (i.e. no mention of supplier) but the cache explicitly calls out `supplier`. 

Should something be renamed based on this discrepancy?